### PR TITLE
change ARES to DARV BAS for VAT number, added some test

### PR DESCRIPTION
**Pull request by jindrichskupa**
I changed ARES register/source to DARV BASIC, where DIC is available and added some test for ICO and DIC.

### CODE REVIEW

File: lib/ruby-ares/address.rb [hunk 1]
```diff
@@ -26,7 +26,7 @@ def initialize(id, street, postcode, city, city_part,
     def to_str
       @city_part = '' if @city == @city_part
 <<EOF
-#{self.street} #{self.street_number}
+#{self.street.present? ? self.street : self.city_part} #{self.street_number}
```
Comment (strzibny) [inline lib/ruby-ares/address.rb]: Can you give me example why this is needed? Also `.present?` is method coming with ActiveSupport, no?

File: lib/ruby-ares/subject.rb [hunk 2]
```diff
@@ -19,11 +20,12 @@ def self.get(ic)
       RubyARES::Parser.get_subject xml
     end
 
-    def initialize(ic, dic, name, status, addresses, updated_at)
+    def initialize(ic, dic, name, status, addresses, updated_at, legal_form)
       @ic, @ico = ic, ic
       @dic = dic
       @name, @company = name
       @status = status
+      @legal_form = legal_form
```
Comment (strzibny) [inline lib/ruby-ares/subject.rb]: This should be assigned at the end of the method in the same order the params are specified.

File: ruby-ares.gemspec [hunk 3]
```diff
@@ -0,0 +1,27 @@
+require 'rubygems/package_task'
```
Comment (strzibny) [inline ruby-ares.gemspec]: If you are creating a separate `.gemspec` file why not to do associated changes to `Rakefile`? It is not nice to manage same code twice.

**Comment by strzibny** [comment]
Thank you! I included a few comments on things I would like to resolve before merging.

File: lib/ruby-ares/address.rb (hunk 1 above)
Comment (jindrichskupa) [inline lib/ruby-ares/address.rb]: You ara true, present? is from rails, so I will change it to nil?. 

Example for this change is: you have address without street (really small villages), usually in the address is village name used for street replacement - http://prntscr.com/93bfdp.

File: lib/ruby-ares/subject.rb (hunk 2 above)
Comment (jindrichskupa) [inline lib/ruby-ares/subject.rb]: OK, I will "fix" it.

**Comment by jindrichskupa** [comment]
You welcome. Just give me a few minutes and I will correct it.

2015-11-16 10:14 GMT+01:00 Josef Strzibny <notifications@github.com>:

> Thank you! I included a few comments on things I would like to resolve
> before merging.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/strzibny/ruby-ares/pull/4#issuecomment-156961945>.
>



-- 
Jindrich Skupa
Tel: +420602807773
E-mail: jindrich.skupa@gmail.com

File: lib/ruby-ares/address.rb (hunk 1 above)
Comment (strzibny) [inline lib/ruby-ares/address.rb]: Okay, I agree. Please change the check to `.nil?`.

**Comment by jindrichskupa** [comment]
I made some additional changes in address.to_str (better name would be just
to_s) and made gemspec as source for Rakefile (dirty, I dont like it). Is
Rakefile required for this purpose ? Please review PR.

Jindrich

2015-11-16 10:24 GMT+01:00 Jindrich Skupa <jindrich.skupa@gmail.com>:

> You welcome. Just give me a few minutes and I will correct it.
>
> 2015-11-16 10:14 GMT+01:00 Josef Strzibny <notifications@github.com>:
>
>> Thank you! I included a few comments on things I would like to resolve
>> before merging.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/strzibny/ruby-ares/pull/4#issuecomment-156961945>.
>>
>
>
>
> --
> Jindrich Skupa
> Tel: +420602807773
> E-mail: jindrich.skupa@gmail.com
>



-- 
Jindrich Skupa
Tel: +420602807773
E-mail: jindrich.skupa@gmail.com

File: lib/ruby-ares.rb [hunk 4]
```diff
@@ -8,5 +8,5 @@
 require 'ruby-ares/parser'
 
 module RubyARES
-  VERSION = '0.0.2'
+  VERSION = '0.0.3'
```
Comment (strzibny) [inline lib/ruby-ares.rb]: Please don't bump the version. I do that in separate commits that I can tag just before releasing.

Comment (jindrichskupa) [inline lib/ruby-ares.rb]: Ok, i will revert this change. Any other notices ? How about change author
to array :))) no just joke ;) do whatever you want.

Thanks

Jindra

Dne pondělí 16. listopadu 2015 Josef Strzibny <notifications@github.com>
napsal(a):

> In lib/ruby-ares.rb
> <https://github.com/strzibny/ruby-ares/pull/4#discussion_r44945143>:
>
> > @@ -8,5 +8,5 @@
> >  require 'ruby-ares/parser'
> >
> >  module RubyARES
> > -  VERSION = '0.0.2'
> > +  VERSION = '0.0.3'
>
> Please don't bump the version. I do that in separate commits that I can
> tag just before releasing.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/strzibny/ruby-ares/pull/4/files#r44945143>.
>


-- 
Jindrich Skupa
Tel: +420602807773
E-mail: jindrich.skupa@gmail.com

Comment (strzibny) [inline lib/ruby-ares.rb]: I have no more objections. I will merge it afterwards.

Comment (jindrichskupa) [inline lib/ruby-ares.rb]: Fixed, you can merge now. Thanks.

**Comment by strzibny** [comment]
Thanks again!

**Comment by strzibny** [comment]
This change is now released as `0.0.3`.

**Comment by jindrichskupa** [comment]
Thank for information, i will change my Gemfiles ;)

2015-11-18 14:13 GMT+01:00 Josef Strzibny <notifications@github.com>:

> This change is now released as 0.0.3.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/strzibny/ruby-ares/pull/4#issuecomment-157708861>.
>



-- 
Jindrich Skupa
Tel: +420602807773
E-mail: jindrich.skupa@gmail.com

**Pull request merged**